packages/back-end/src/integrations/SqlIntegration.ts

```diff
@@ -750,16 +752,24 @@ export default abstract class SqlIntegration
     query: string,
     templateVariables?: TemplateVariables
   ): string {
-    return this.getTestQuery(query, templateVariables, 1);
+    return this.getTestQuery(
+      query,
+      templateVariables,
+      DEFAULT_TEST_QUERY_DAYS,
+      1
+    );
   }
 
   getTestQuery(
     query: string,
     templateVariables?: TemplateVariables,
+    testDays?: number,
     limit: number = 5
   ): string {
     const startDate = new Date();
-    startDate.setDate(startDate.getDate() - IMPORT_LIMIT_DAYS);
+    startDate.setDate(
+      startDate.getDate() - (testDays ?? DEFAULT_TEST_QUERY_DAYS)
```

Comment:
Should you validate that `testDays` is positive?

Reply:
There's validation in the front-end which I think is good enough for now.